code/modules/xenoarcheaology/finds/strange_rock.dm

```diff
@@ -30,7 +30,7 @@
 	if(IS_WELDER(I))
 		var/obj/item/weldingtool/W = I
 		if(W.isOn())
-			if(W.remove_fuel(2))
+			if(W.weld(2))
```

Comment:
If I add the user arg it would create duplicate messages. That's why it wasn't passed before and isn't now either.
Also notice that there's no return TRUE if the welder isn't on, so it can call its after attack when off.